rooms 추가 구조 개선 및 리팩토링

yxxjxxeee opened the pull request:
## #️⃣ 연관된 이슈
- #15 

## 📝 작업 내용
> Rooms 모듈의 책임 분리와 구조를 개선하여 유지보수성과 확장성을 높였습니다.
**DTO**: class-transformer 적용 
**Controller**: 응답 DTO 변환(resDto) 책임 컨트롤러로 이동 
**Pagination**: 서비스 로직을 클래스로 변경, cursor와 take 활용하여 필요한 범위만 조회 
**Room Entity**: 비밀번호 검증 로직 엔티티 내부로 이동 (비즈니스 규칙 캡슐화) 
**Mapper**: 변환 책임 분리 
**Constants**: 상수 정리 
**DB 스키마**: Room isPrivate 기본값 제거 
**IRoomRepository** 인터페이스 도입 
**PasswordEncoderService**: DI에서 단순 유틸 객체로 변경 
**Room.toPersistence**: 명시적 반환 타입 추가하여 DB 스키마 안전성 강화

yxxjxxeee committed:
refactor: (#15) rooms 추가 구조 개선 및 Entity, Controller, DTO 리팩토링

yxxjxxeee committed:
refactor: (#15) service 의존성을 인터페이스 기반 DI로 변경

kimjiwonpg98 commented on src/rooms/entities/rooms.entity.ts:
```diff
@@ -1,80 +1,148 @@
+  ): Promise<Room> {
+    if (!isPrivate && plainPassword) {
+      throw new BadRequestException(ERROR_MESSAGES.PUBLIC_ROOM_NO_PASSWORD);
+    }
```
비공개방이 아니면 password를 null로 넣을꺼라면 에러가 아니라 무시하는 방법도 있을 것 같네요

yxxjxxeee replied:
좋은 것 같습니다. 에러 대신 password를 그냥 null로 처리하는 방법이 더 자연스러울 것 같습니다!

kimjiwonpg98 commented on src/rooms/rooms.service.ts:
```diff
@@ -1,80 +1,66 @@
-import { PasswordEncoderService } from 'src/common/hashing/password-encoder.service';
-import { FindRoomsQuery } from './rooms.interface';
 import { Room } from './entities/rooms.entity';
+import { PasswordEncoderService } from 'src/common/hashing/password-encoder.service';
```
PasswordEncoderService 같은 경우는 단순 유틸 함수처럼 사용하는거같은데
DI까지 해서 쓰는건 제가 느끼기엔 너무 DI를 헀다는 생각이 드네요

단순 객체로 만들어서 쓰는것도 고민해보면 좋을 것 같습니다

yxxjxxeee replied:
제 생각에도 DI로 주입하는 건 과한 것 같아서 말씀하신 것처럼  단순 유틸 성격이므로 모듈화된 유틸 객체로 변경하여 사용하는 쪽으로 수정했습니다!

rrgks6221 commented on src/rooms/constants/room.constant.ts:
```diff
@@ -0,0 +1,15 @@
+export const ERROR_MESSAGES = {
```
결국 백엔드에서 사용자에게 노출하는 메시지를 넘겨주기로 결정됐나 보군요. ㅠ

yxxjxxeee replied:
프론트엔드 분과 다시 상의한 결과 백엔드에서는 사용자에게 보여줄 메시지를 직접 전달하지 않고 에러 코드를 전달하기로 했습니다.
따라서 메시지 대신 ERROR_CODES를 전달하도록 수정했습니다!

rrgks6221 commented on src/rooms/entities/mappers/rooms.mapper.ts:
```diff
@@ -0,0 +1,27 @@
+    );
+  }
+
+  static toPersistence(room: Room) {
```
toPersistence 메서드에도 응답타입을 지정하면 DB 스키마가 변경됐을 때 이점이 있을거같아요

```suggestion
  static toPersistence(room: Room): PrismaRoom {
```

yxxjxxeee replied:
좋은 것 같습니다!

rrgks6221 commented on src/rooms/entities/rooms.entity.ts:
```diff
@@ -1,80 +1,148 @@
+
+  hasPassword(): boolean {
+    return this.hashedPassword !== null;
+  }
```
두 동작은 메서드가 아닌 단순히 getter만으로도 충분하지 않을까싶은 의견이 있어요

메서드란 어떠한 행위를 표현하거나 인자를 통해 계산, 로직이 들어갈 때 의미가 있는데 단순한 객체 속성을 가져오는 정도로 보여서요

yxxjxxeee replied:
말씀하신 것처럼 단순 속성 반환 정도라면 굳이 메서드로 만들기보다는 getter로 정의하는 것이 더 직관적인 것 같습니다!

yxxjxxeee committed:
refactor: (#15) Room 비밀번호 처리 및 getter 개선 반영

- PasswordEncoderService를 DI에서 단순 유틸 객체로 변경
- Room 엔티티의 단순 속성은 메서드 대신 getter 사용
- 공개 방으로 변경 시 비밀번호를 null로 처리하도록 수정
- Room.toPersistence에 명시적 반환 타입 추가하여 DB 스키마 안전성 강화
